### PR TITLE
Fix an issue in Edgeview not getting App interface IP address

### PR DESCRIPTION
**Pull request by naiming-zededa**
- due to the change of multiple port in zedrouter, the
  AppNetAdapterList.AllocatedIPv4Addr was changed to AssignedAddresses,
  do the same change
- update the vendor files for pkg/pillar in pkg/edgeview

### CODE REVIEW

**Review by milan-zededa — APPROVED** [review]
Thanks for fixing this.
We should mark this with the stable label and backport this to 13.4-stable

File: pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go [hunk 1]
```diff
@@ -29,6 +29,100 @@ type AssignableAdapters struct {
...
+func (iobe *IOBundleError) Empty() bool {
+	if iobe.Errors == nil || len(iobe.Errors) == 0 {
+		return true
+	}
+
+	return false
+}
+
+// HasErrorByType returns true if error of the same type is found
+func (iobe *IOBundleError) HasErrorByType(e error) bool {
```
Comment (christoph-zededa) [inline pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go:90]: This is already in master: https://github.com/lf-edge/eve/blob/master/pkg/pillar/types/assignableadapters.go#L90

I don't understand why GH is showing this; yesterday @rucoder had a similar issue ...

Comment (milan-zededa) [reply pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go:90]: But this is a copy of that file vendored under edgeview package.
Or do you mean that we miss some recent change here?

Comment (christoph-zededa) [reply pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go:90]: ah sorry, my mistake; then everything is fine.

Comment (christoph-zededa) [reply pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go:90]: Can't edgeview just use `replace` in go.mode?

Comment (milan-zededa) [reply pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go:90]: @deitch can write an essay on why `replace` is evil :)

Comment (OhmSpectator) [reply pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go:90]: > Can't edgeview just use `replace` in go.mod?

Do we use the old version there?... So the PR should be fixed?

I remember we had a reason to put our code under vendor (though I need to remember it exactly), but it still looks awful and has to be at least automated from the perspective of updating/checking in PRs...

Comment (deitch) [reply pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go:90]: > write an essay

I can, you won't want to read it, though. :-)

Basically don't do it, except for local circumstances, like "I am developing something and I need to use my local copy of this other dependency, just to see if it all works together, but I will not commit it that way."

I don't understand what problem edgeview is trying to solve that it needs `replace`? Or is it the issue that `pkg/edgeview` and `pkg/pillar` each build independently, but there is common code to both, so you want to reuse one from the other?

Comment (christoph-zededa) [reply pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go:90]: Just to not having to create these PRs and make it automatic - is there another way?

Comment (deitch) [reply pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go:90]: Libraries, like [eve-libs](http://github.com/lf-edge/eve-libs/), or some sort of tooling that auto-generates, but those are painful. Pretty soon you are building bazel.

Comment (OhmSpectator) [reply pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go:90]: I'm still confused about what we are doing with this PR. Should it be fixed?

Comment (christoph-zededa) [reply pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go:90]: From my side it is fine then.

**Review by OhmSpectator — APPROVED** [review]

File: pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go (hunk 1 above)
Comment (shjala) [reply pkg/edgeview/vendor/github.com/lf-edge/eve/pkg/pillar/types/assignableadapters.go:90]: > > write an essay
> 
> I can, you won't want to read it, though. :-)

Meanwhile Eden's go.mod :

**Review by OhmSpectator — APPROVED** [review]

**Comment by OhmSpectator** [comment]
@naiming-zededa we still need to backport it to 13.4.

**Pull request merged**